base/server/src/com/netscape/cmscore/systemd/SystemdStartupNotifier.java

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package com.netscape.cmscore.systemd;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.netscape.certsrv.base.IConfigStore;
+import com.netscape.cmscore.apps.SubsystemListener;
+
+public class SystemdStartupNotifier extends SubsystemListener {
+
+    String pid;
+
+    @Override
+    public void init(IConfigStore config) throws Exception {
+
+        logger.info("SystemdStartupNotifier: Initialization");
+
+        // https://stackoverflow.com/questions/35842/how-can-a-java-program-get-its-own-process-id
+        RuntimeMXBean runtime = ManagementFactory.getRuntimeMXBean();
+
+        String jvmName = runtime.getName();
+        logger.info("SystemdStartupNotifier: - JVM name: " + jvmName);
+
+        int index = jvmName.indexOf("@");
+        if (index < 0) {
+            throw new Exception("Unable to get PID from JVM name: " + jvmName);
+        }
+
+        pid = jvmName.substring(0, index);
```

Comment:
Default behaviour of `systemd-notify(1)` is to use the pid of the of the process that invoked `systemd-notify`.  Omitting the `--pid` option, and this code for finding out what the pid is, would simplify the implementation.

Reply:
The `--pid` option was added per @tiran's suggestion, but if it works without it that would be better. Please take a look at the updated patch. Thanks!

Reply:
@edewata I will test this patch in the next day or so.